packages/core/helper-plugin/src/components/SearchURLQuery/index.js

```diff
@@ -9,22 +9,22 @@ import { useTracking } from '../../features/Tracking';
 import useQueryParams from '../../hooks/useQueryParams';
 
 const SearchURLQuery = ({ label, placeholder, trackedEvent, trackedEventDetails }) => {
-  const wrapperRef = useRef(null);
+  const inputRef = useRef(null);
   const iconButtonRef = useRef(null);
 
   const [{ query }, setQuery] = useQueryParams();
+
   const [value, setValue] = useState(query?._q || '');
   const [isOpen, setIsOpen] = useState(!!value);
+
   const { formatMessage } = useIntl();
   const { trackUsage } = useTracking();
 
   const handleToggle = () => setIsOpen((prev) => !prev);
 
   useLayoutEffect(() => {
-    if (isOpen) {
-      setTimeout(() => {
-        wrapperRef.current.querySelector('input').focus();
-      }, 0);
+    if (isOpen && inputRef.current) {
+      inputRef.current.focus();
```

Comment:
Have you tested whether the i18n locale switch in the CM list view still works? I remember we had to add the setTimeout because of rendering race conditions.

Reply:
When I switch the locale in the CM list view, the component changes from it's input back to its button – is this right? I've never noticed something like this.

Reply:
Not sure I understand :) When you change the locale in the CM list-view: 1) the URL param should change, the page should re-render and display only entries from the new locale and the select should display the updated locale as far as I remember.

Reply:
Nothing seems broken 👍🏼